src/utils/cache.ts

```diff
@@ -0,0 +1,46 @@
+import os from 'os';
+import path from 'path';
+import { env } from 'process';
+import { fetchLatestGitHubRelease } from './version';
+import { readFileWithStats } from './file';
+import * as logger from './logger';
+import { mkdir, writeFile } from 'fs/promises';
+
+export function getCacheDir(): string {
+  return path.join(
+    env.XDG_CACHE_HOME || path.join(os.homedir(), '.cache'),
+    'cio-sdk-tools'
+  );
+}
+
+// to avoid hitting GitHub api rate limits, we'll cache the latest version for an hour
+export async function fetchCachedLatestVersion(
+  packageName: string
+): Promise<string | undefined> {
+  const cacheDir = getCacheDir();
+  const cacheKey = `${packageName}.latest`;
+  const cachePath = path.join(cacheDir, cacheKey);
+
+  const results = readFileWithStats([cachePath]);
+  if (results) {
+    const r = results[0];
+    if (r.lastUpdated && r.lastUpdated >= Date.now() - 60 * 60 * 1000) {
```

Comment:
Can we make the comparison a bit more clear by using variable here instead? e.g.

```
    const cacheDurationMs = 60 * 60 * 1000;
    if (r.lastUpdated && r.lastUpdated >= Date.now() - cacheDurationMs) {
```

or

```
    const oneHourCacheTime = 60 * 60 * 1000;
    if (r.lastUpdated && r.lastUpdated >= Date.now() - oneHourCacheTime) {
```